dask_cuda/device_host_file.py

```diff
@@ -115,7 +116,11 @@ def __sizeof__(self):
 
     def __reduce_ex__(self, protocol):
         header, frames = device_serialize(self)
-        frames = [f.obj for f in frames]
+        # Since pickle cannot handle memoryviews, we convert them
+        # to NumPy arrays (zero-copy).
+        frames = [
+            (numpy.asarray(f) if isinstance(f, memoryview) else f) for f in frames
+        ]
```

Comment:
What is happening here? AKA how did this ever work if this change is needed?

Reply:
It assumed that `[f.obj for f in frames]` would convert each frame to a numpy array. This is not the case after https://github.com/rapidsai/cudf/pull/11447

Reply:
Thanks